test/sql/catalog/test_if_not_exists.test

```diff
@@ -14,6 +14,15 @@ CREATE TABLE IF NOT EXISTS integers2 AS SELECT 42
 statement ok
 CREATE TABLE IF NOT EXISTS integers2 AS SELECT 42
 
+# similar to Postgres, treat as NOOP if the entry exists but the type is different
+statement ok
+CREATE VIEW IF NOT EXISTS integers2 AS SELECT 42
```

Comment:
@Mytherin how do you feel about this?
Postgres has this behavior (minus a warning/notice message)

I'd say this should fail because a VIEW by this name doesn't exist, which sounds like what the user expects when this passes to me

Reply:
Also note that this (not throwing an exception) is the current behavior of DuckDB as well, there are no actual behavioral changes by this PR except for fixing the issue where the version is no longer incremented on these NOOP commands.